runtime/autoload/dist/ft.vim

```diff
@@ -465,7 +465,7 @@ export def ProtoCheck(default: string)
     # recognize Prolog by specific text in the first non-empty line
     # require a blank after the '%' because Perl uses "%list" and "%translate"
     var lnum = getline(nextnonblank(1))
-    if lnum =~ '\<prolog\>' || lnum =~ '^\s*\(%\+\(\s\|$\)\|/\*\)' || lnum =~ ':-'
+    if lnum =~ '\<prolog\>' || lnum =~ '(^\s*(:-\|%\|\/\*))\|.\s*$'
```

Comment:
This pattern doesn't make sense. It will match any non-empty line, because `.\s*` can match any character at the end of a line.

Reply:
I left it in case there were trailing whitespaces, just in case. Is that a bad idea?

Reply:
It will match any non-empty line:
```vim
:echo 'foobar' =~ '(^\s*(:-\|%\|\/\*))\|.\s*$'
```
prints 1.

Reply:
If you want to match a `.` at the end of the line. You should use `\.\s*` instead. But that doesn't pass the tests, which means the pattern is still wrong :(

Reply:
Yea, you're right. I will work on fixing it right away.

Reply:
Writing this:

```viml
let regex = '^\(:-\|%\|\/\*\)\|\.$'

let pattern1 = ':-module(test/1,'
let pattern2 = '% comment'
let pattern3 = '/* multiline comment'
let pattern4 = 'rule(test, 1.7).'
let patternX = 'a'

if pattern1 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern2 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern3 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if pattern4 =~ regex
  echo 'OK'
else
  echo 'ERROR'
endif

if patternX =~ regex
  echo 'ERROR'
else
  echo 'OK'
endif
```

All tests are passing now. I forgot that in Vim regexes grouping is with `\(` and not just `(`. I'm pushing the fix now. Thanks @zeertzjq for the feedback!

Reply:
@chrisbra how can I find out which patch number I should use?

Reply:
@ignamartinoli He will add a patch number when he commit a patch to the master branch. You don't need to add a patch number by yourself.

Reply:
@k-takata thanks for the info. I think I was able to fix it now